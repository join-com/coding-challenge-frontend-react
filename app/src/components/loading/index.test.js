import React from 'react'
import LoadingHCIComponent from './index'
import renderer from 'react-test-renderer'
import { shallow, render } from 'enzyme'

const testComponent = props => <div>Test Component</div>

test('Should render loading component', () => {
  const Loader = LoadingHCIComponent(testComponent)
  const component = render(<Loader isLoading={true} />)
  expect(component).toMatchSnapshot()
})

test('Should render provided component when not in loading', () => {
  const Loader = LoadingHCIComponent(testComponent)
  const component = render(<Loader isLoading={false} />)
  expect(component).toMatchSnapshot()
})
