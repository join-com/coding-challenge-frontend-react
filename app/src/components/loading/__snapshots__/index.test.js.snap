// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render loading component 1`] = `
<div
  class="d-flex justify-content-center"
>
  <div
    class="spinner-border"
    role="status"
  >
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;

exports[`Should render provided component when not in loading 1`] = `
<div>
  Test Component
</div>
`;
